src/transformers/models/rt_detr/configuration_rt_detr.py

```diff
@@ -0,0 +1,338 @@
+            based on the predictions from the previous layer.
+        is_encoder_decoder (`bool`, *optional*, defaults to `True`):
+            Whether the architecture has an encoder decoder structure.
+        matcher_alpha (`float`, *optional*, defaults to 0.25):
+            Parameter alpha used by the Hungarian Matcher.
+        matcher_gamma (`float`, *optional*, defaults to 2.0):
+            Parameter gamma used by the Hungarian Matcher.
+        matcher_class_cost (`float`, *optional*, defaults to 2.0):
+            The relative weight of the class loss used by the Hungarian Matcher.
+        matcher_bbox_cost (`float`, *optional*, defaults to 5.0):
+            The relative weight of the bounding box loss used by the Hungarian Matcher.
+        matcher_giou_cost (`float`, *optional*, defaults to 2.0):
+            The relative weight of the giou loss of used by the Hungarian Matcher.
+        use_focal_loss (`bool`, *optional*, defaults to `True`):
+            Parameter informing if focal focal should be used.
+        auxiliary_loss (`bool`, *optional*, defaults to `True`):
+            Whether auxiliary decoding losses (loss at each decoder layer) are to be used.
+        focal_loss_alpha (`float`, *optional*, defaults to 0.75):
+            Parameter alpha used to compute the focal loss.
+        focal_loss_gamma (`float`, *optional*, defaults to 2.0):
+            Parameter gamma used to compute the focal loss.
+        weight_loss_vfl (`float`, *optional*, defaults to 1.0):
+            Relative weight of the varifocal loss in the object detection loss.
+        weight_loss_bbox (`float`, *optional*, defaults to 5.0):
+            Relative weight of the L1 bounding box loss in the object detection loss.
+        weight_loss_giou (`float`, *optional*, defaults to 2.0):
+            Relative weight of the generalized IoU loss in the object detection loss.
+        eos_coefficient (`float`, *optional*, defaults to 0.0001):
+            Relative classification weight of the 'no-object' class in the object detection loss.
+
+    Examples:
+
+    ```python
+    >>> from transformers import RTDetrConfig, RTDetrModel
+
+    >>> # Initializing a RT-DETR configuration
+    >>> configuration = RTDetrConfig()
+
+    >>> # Initializing a model (with random weights) from the configuration
+    >>> model = RTDetrModel(configuration)
+
+    >>> # Accessing the model configuration
+    >>> configuration = model.config
+    ```"""
+
+    model_type = "rt_detr"
+    attribute_map = {
+        "hidden_size": "d_model",
+        "num_attention_heads": "encoder_attention_heads",
+    }
+
+    def __init__(
+        self,
+        initializer_range=0.01,
+        layer_norm_eps=1e-5,
+        batch_norm_eps=1e-5,
+        # backbone
+        use_timm_backbone=True,
+        backbone_config=None,
+        backbone="resnet50d",
+        use_pretrained_backbone=True,
+        backbone_kwargs=None,
+        # encoder HybridEncoder
+        d_model=256,
```

Comment:
@SangbumChoi 
`d_model` seems to be dedicated to `HybridEncoder`, however it is being use in `RTDetrDecoderLayer` as well, and this will cause problem for using `rtdetr_r101vd`.

Referring to
https://github.com/lyuwenyu/RT-DETR/blob/2b88d5d53bcbfbb70329bc9c007fdf7e76cf90dc/rtdetr_pytorch/configs/rtdetr/rtdetr_r101vd_6x_coco.yml#L13-L22

`rtdetr_r101vd` will have `d_model` set to `386` for `HybridEncoder`, but keep using `256` for `RTDetrDecoderLayer`

Reply:
I think we should add additional configuration for the hybridencoder which is `config.e_model`

Reply:
Seems it will be more clear if rename them to `encoder_hidden_dim` and `decoder_hidden_dim`.

Reply:
it is true. However, d_model is not only related to decoder_hidden_dim so I will remain the name as-is.

Reply:
d_model -> dimension of model

Reply:
@SangbumChoi 

`RTDetrCSPRepLayer` and `RTDetrEncoderLayer` used in `RTDetrHybridEncoder` will directly use `config.d_model`, they should be using `encoder_hidden_dim` as well.
(Better double check are there more placing directly using `config.d_model` which should use `encoder_hidden_dim` instead)

Reply:
@SangbumChoi Could you re-check with the latest version?